### PR TITLE
Add check for SystemRequirements,

### PR DESCRIPTION
**Pull request by nturaga**
SystemRequirements in the DESCRIPTION file provide dependencies
which are usually not fulfilled by the bioconductor-skeleton.py.

These usually represent external dependencies.

Take for instance the package 'rsbml', which has the SystemRequirement,
libsbml (==5.10.2). This is not fulfilled by this script, and should
throw a warning to the user that there is a recipe for libsbml is also
needed.

There are many packages which have such SystemRequirements with
Bioconductor, as shown in this gist.

    https://gist.github.com/nturaga/39185116b8681264e7405b868e00d405

If there are suggestions from the Bioconda team to standardizing the format of
SystemRequirements, I could try to get them standardized across
Bioconductor packages, so we have a better chance of fulfilling them
automatically.

I'm cc-ing @mtmorgan, and @vobencha to help answer add context if
needed.

The PR leaves an output like this,

```
(base) ~/D/bioconda-recipes ❯❯❯ bioconda-utils bioconductor-skeleton recipes config.yml rsbml
15:08:36 BIOCONDA INFO Making recipe for: rsbml
15:08:36 BIOCONDA DEBUG rsbml==2.40.0, BioC==3.8
15:08:37 BIOCONDA INFO Using tarball from https://bioconductor.org/packages/3.8/bioc/src/contrib/rsbml_2.40.0.tar.gz
15:08:37 BIOCONDA INFO creating recipes/bioconductor-rsbml
15:08:37 BIOCONDA INFO BioConductor dependency: name="BiocGenerics" version=">=0.28.0,<0.30.0"
15:08:37 BIOCONDA INFO            R dependency: name="R" version=">=2.6.0"
15:08:37 BIOCONDA INFO BioConductor dependency: name="graph" version=">=1.60.0,<1.62.0"
15:08:37 BIOCONDA WARNING The 'SystemRequirements' libsbml (==5.10.2) are needed by the recipe for the package to work as intended.
15:08:37 BIOCONDA INFO Wrote recipe in recipes/bioconductor-rsbml
```

### CODE REVIEW

**Comment by dpryan79** [comment]
Great idea! We should also add these as a comment line in the `meta.yaml`.

Regarding standardizing that, something like `package (==optional.version.pinning), some_other_package` would be great. I imagine there's no way to really control the vocabulary there, but providing package submitters/maintainers with a list of the most common dependencies and how to denote them would help (e.g., `gsl` should be used to denote the GNU Scientific Library). It's going to be really hard to parse that if comments (e.g., "gsl. Note: users should have GSL installed. Windows"). There should also be a distinction between what's actually required for base functionality and what is optional, though that'd require adding entries.

@bgruening Any other thoughts about this?

**Comment by bgruening** [comment]
Any additional annotation would be great. I'm also skeptical that we can agree on a nomenclature but any step in this direction is helpful and when we can move these comments in the recipe it gets easier to fix failing recipes. We need to see if we really want to follow the strict pinning. Bioc is not harmonizing the recommendations or do you @nturaga? What I mean is there only one GSL version recommended across all recipes, or can this be multiple?

So I guess I have nothing to add besides excitement :)

**Comment by dpryan79** [comment]
OK I'll merge this in and then add that in as a comment then same way we handle `Suggests:`. We can kick the standardization can down the road then :)
